main.py

```diff
@@ -0,0 +1,52 @@
+import random
+
+def pick_random_names_from_list(user_input):
+    """
+    Function takes names in a string and returns a specified number of random results
+
+    Parameters
+    -----
+    user_input : string
+
+    returns
+    -----
+    result : string
+    """
+    name_list = user_input.split(',')
+
+    while True:
+        shortlist_count = input('Enter the number of names you want selected: ')
+        if shortlist_count.isdigit() and int(shortlist_count) > 0:
+            shortlist_count = int(shortlist_count)
+            if shortlist_count <= len(name_list):
+                break
+            else:
+                print(f'You entered {shortlist_count}, but there are only {len(name_list)} names in the list. Please enter a smaller number.')
+        else:
+            print(f'You entered {shortlist_count}, but you must select a valid number of 1 or higher')
```

Comment:
Could do with using a try and except to possibly make the code a bit more concise.